Backport of:

From df14e6c0b9592cbb24d5381dfc6106b14f915e75 Mon Sep 17 00:00:00 2001
From: Simon Bünzli <zeniko@gmail.com>
Date: Wed, 19 Feb 2014 07:42:52 +0000
Subject: Fix Savannah bug #41590.

Origin: ubuntu, http://bazaar.launchpad.net/~ubuntu-branches/ubuntu/precise/freetype/precise-security/view/head:/debian/patches-freetype/savannah-bug-41590.patch
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=798620

* src/type1/t1load.c (parse_encoding): Protect against invalid
number.
---
#diff --git a/ChangeLog b/ChangeLog
#index 945962f..c1d5aea 100644
#--- a/ChangeLog
#+++ b/ChangeLog
#@@ -4,6 +4,13 @@
# 
# 	* src/autofit/hbshim.c: Include `afglobal.h' and `aftypes.h'.
# 
#+2014-02-19  Simon Bünzli  <zeniko@gmail.com>
#+
#+	Fix Savannah bug #41590.
#+
#+	* src/type1/t1load.c (parse_encoding): Protect against invalid
#+	number.
#+
# 2014-02-12  Dave Arnold  <darnold@adobe.com>
# 
# 	[cff] Optimize by using `FT_MulDiv'.
Index: freetype-2.4.2/src/type1/t1load.c
===================================================================
--- freetype-2.4.2.orig/src/type1/t1load.c
+++ freetype-2.4.2/src/type1/t1load.c
@@ -1249,6 +1249,13 @@
           {
             charcode = (FT_Int)T1_ToInt( parser );
             T1_Skip_Spaces( parser );
+
+            /* protect against invalid charcode */
+            if ( cur == parser->root.cursor )
+            {
+              parser->root.error = FT_Err_Unknown_File_Format;
+              return;
+            }
           }
 
           cur = parser->root.cursor;
